apps/api/src/providers/provider.regression.ts

```diff
@@ -1,23 +1,55 @@
+import { ChannelTypeEnum, EmailProviderIdEnum } from '@novu/shared';
 import { UserSession } from '@novu/testing';
 import { expect } from 'chai';
 
-import { createRegressionNotificationTemplate } from './helpers';
+import { checkProviderIntegration, createProviderIntegration, createRegressionNotificationTemplate } from './helpers';
+import { getProviderSecrets } from './secrets';
 
-const providerId = 'sendgrid';
+const providers = [EmailProviderIdEnum.SendGrid];
 
 let session: UserSession;
 
-describe('Regression test - Provider', () => {
+describe('Regression test - Providers', () => {
+  before(async () => {
+    session = new UserSession();
+    await session.initialize();
+  });
+
   describe('Email channel', () => {
-    describe(`Provider ${providerId}`, () => {
-      beforeEach(async () => {
-        session = new UserSession();
-        await session.initialize();
-        await createRegressionNotificationTemplate(session, providerId);
-      });
+    const channel = ChannelTypeEnum.EMAIL;
+
+    providers.forEach((providerId) => {
+      describe(`Provider ${providerId}`, () => {
+        let secrets;
+
+        before(async () => {
+          await createProviderIntegration(session, providerId, channel);
+        });
+
+        beforeEach(async () => {
+          secrets = getProviderSecrets(providerId);
+          await createRegressionNotificationTemplate(session, providerId);
+        });
+
+        describe('Setting up', () => {
+          it('should have the proper integration active and enabled', async () => {
+            const integration = await checkProviderIntegration(session, providerId);
+
+            expect(integration).to.deep.include({
+              _environmentId: session.environment._id,
+              _organizationId: session.organization._id,
+              providerId,
+              channel,
+              credentials: { apiKey: secrets.apiKey, secretKey: secrets.secretKey },
+              active: true,
+              deleted: false,
+            });
+          });
 
-      it('should create notification template properly', async () => {
-        expect(1).to.eql(0);
+          it('should create notification template properly', async () => {
+            expect(1).to.eql(0);
```

Comment:
I am leaving this on purpose until more work is created as a flag that is not work done. Also I am PRing against a Draft PR as a safety measure.